app/controllers/ops_controller/ops_rbac.rb

```diff
@@ -902,13 +902,45 @@ def rbac_group_get_details(id)
   def rbac_group_right_tree(selected_nodes)
     case @sb[:active_rbac_group_tab]
     when 'rbac_customer_tags'
-      @tags_tree = TreeBuilderTags.new(:tags_tree,
-                                       :tags,
-                                       @sb,
-                                       true,
-                                       :edit    => @edit,
-                                       :filters => @filters,
-                                       :group   => @group)
+      cats = Classification.categories.find_all do |c|
+        c if c.show || !%w[folder_path_blue folder_path_yellow].include?(c.name)
+      end
+      cats.sort_by! { |t| t.description.try(:downcase) } # Get the categories, sort by description
+      cats.delete_if { |c| c.read_only? || c.entries.empty? } # Remove categories that are read only or have no entries
+      @tags = cats.map do |cat|
+        {
+          :id          => cat.id,
+          :description => cat.description,
+          :singleValue => cat.single_value,
+          :values      => cat.entries.sort_by { |e| e[:description.downcase] }.map do |entry|
+            { :id => entry.id, :description => entry.description }
+          end
+        }
+      end
+      filters = @edit&.dig(:new)&.dig(:filters) || @filters
+      assigned_tags = Tag.where(:name => filters.flatten).map do |tag|
+        {
+          :description => tag.category.description,
+          :id          => tag.category.id,
+          :values      => [{:id => tag.classification.id, :description => tag.classification.description}]
+        }
+      end
+
+      assigned_tags.each_with_object([]) do |tag, arr|
+        existing_tag = arr.find { |item| item[:id] == tag[:id] }
+        if existing_tag
+          existing_tag[:values].push(*tag[:values])
+        else
+          arr << tag
+        end
+      end
+
+      assigned_tags.uniq! { |tag| tag[:id] }
+      @tags = {:tags => @tags, :assignedTags => assigned_tags, :affectedItems => [@group.id]}
```

Comment:
Recursively setting a variable increases code complexity, it counts as a double if you do it for an instance variable.

```ruby
@tags = { :tags => @tags }
```